src/components/SqlEditor.tsx

```diff
@@ -105,26 +116,33 @@ export const SQLEditor = (props: SQLEditorProps) => {
   };
 
   return (
-    <div className={styles.Common.wrapper}>
-      <a
-        onClick={() => onToggleExpand()}
-        className={styles.Common.expand}
-        data-testid={selectors.components.QueryEditor.CodeEditor.Expand}
-      >
-        <i className={`fa fa-${expand.icon}`}></i>
-      </a>
-      <CodeEditor
-        aria-label="SQL"
-        height={expand.height}
-        language="sql"
-        value={query.rawSql || ''}
-        onSave={onSqlChange}
-        showMiniMap={false}
-        showLineNumbers={true}
-        onBlur={(text) => onChange({ ...query, rawSql: text })}
-        onEditorDidMount={(editor: any) => handleMount(editor)}
-      />
-    </div>
+    <>
+      { app === CoreApp.Explore &&
+        <div className={'gf-form ' + styles.QueryEditor.queryType}>
+          <QueryTypeSwitcher queryType={queryType} onChange={queryType => saveChanges({ queryType })} sqlEditor />
+        </div>
+      }
+      <div className={styles.Common.wrapper}>
+        <a
+          onClick={() => onToggleExpand()}
+          className={styles.Common.expand}
+          data-testid={selectors.components.QueryEditor.CodeEditor.Expand}
+        >
+          <i className={`fa fa-${expand.icon}`}></i>
+        </a>
+        <CodeEditor
+          aria-label="SQL"
+          height={expand.height}
+          language="sql"
+          value={query.rawSql || ''}
```

Comment:
In what cases `rawSql` in falsy? the typing says it's `string`

Reply:
When the editor first loads (sql or builder), it only contains `datasource` and `refId`. We could mark all the properties as optional to match this behavior, but it's only true before the first edit to the query. After that all of the required properties are set.

In this case the `value` prop will be fine without the `|| ''` so I've removed it